client/config/env.js

```diff
@@ -12,16 +12,20 @@ if (!NODE_ENV) {
   );
 }
 
+const ENV_FILE = process.env.ENV_FILE;
+
 // https://github.com/bkeepers/dotenv#what-other-env-files-can-i-use
-const dotenvFiles = [
-  `${paths.dotenv}.${NODE_ENV}.local`,
-  // Don't include `.env.local` for `test` environment
-  // since normally you expect tests to produce the same
-  // results for everyone
-  NODE_ENV !== "test" && `${paths.dotenv}.local`,
-  `${paths.dotenv}.${NODE_ENV}`,
-  paths.dotenv,
-].filter(Boolean);
+const dotenvFiles = ENV_FILE
+  ? [paths.dotenv.replace(".env", ENV_FILE)]
+  : [
+      `${paths.dotenv}.${NODE_ENV}.local`,
+      // Don't include `.env.local` for `test` environment
+      // since normally you expect tests to produce the same
+      // results for everyone
+      NODE_ENV !== "test" && `${paths.dotenv}.local`,
+      `${paths.dotenv}.${NODE_ENV}`,
+      paths.dotenv,
+    ].filter(Boolean);
```

Comment:
If we wanted to be precise, we should change this to:

```suggestion
  : [paths.dotenv];
```

(Since `.env.local` etc would only be applied by the `client/`, but not the rest of yari.)

Reply:
Yes please! Keeps it simpler, consistent, and removes code from the mega webpack config!

We also need to remove the comment on L17 which refers to this behaviour.

Reply:
Could we also simplify the lines below: https://github.com/caugner/yari/blob/61c83eb14d556cc748bee89d434a81dafc96dd07/client/config/env.js#L30-L44

We don't seem to utilize the functionality of dotenv-expand, nor do we use the module in the rest of the client, and we only use a single dotenv file at a time, so we can remove the array and `forEach`.

Reply:
Good point, done in 82346ba97522c5381f48fd092191c043d6af9c8c.